packages/sanity/src/core/config/prepareConfig.ts

```diff
@@ -500,16 +499,14 @@ function resolveSource({
       config,
     }),
     document: {
-      actions: (partialContext) => {
-        const actions = resolveConfigProperty({
+      actions: (partialContext) =>
+        resolveConfigProperty({
           config,
           context: {...context, ...partialContext},
           initialValue: initialDocumentActions,
           propertyName: 'document.actions',
           reducer: documentActionsReducer,
-        })
-        return getStartInCreateSortedActions(actions)
-      },
+        }),
```

Comment:
Getting rid of this sort hack is honestly good. This feature no longer messes with actions (unless the document is linked with Create, in which case it doesnt render actions menu at all, only the Unlink actions like before)